jit/expressionjit.h

```diff
@@ -0,0 +1,105 @@
+#include <queue>
+#include <utility>
+
+#include "compileoperator.h"
+#include "returnedcolumn.h"
+#include "arithmeticcolumn.h"
+
+using namespace execplan;
+
+namespace msc_jit
+{
+static JIT& getJITInstance()
+{
+  static JIT jit;
+  return jit;
+}
+
+class CompiledColumn : public execplan::ReturnedColumn
+{
+ private:
+  CompiledOperator compiledOperator;
+
+ public:
+  CompiledColumn()
+  {
+  }
+  CompiledColumn(CompiledOperator compiledOperator, ReturnedColumn* expression)
+   : ReturnedColumn(*expression), compiledOperator(std::move(compiledOperator))
+  {
+  }
+
+  inline CompiledColumn* clone() const override
+  {
+    return new CompiledColumn(*this);
+  }
+  bool hasWindowFunc() override
+  {
+    return false;
+  }
+
+ public:
+  int64_t getIntVal(rowgroup::Row& row, bool& isNull) override
+  {
+    return compiledOperator.compiled_function_int64(row.getData(), isNull);
+  }
+  uint64_t getUintVal(rowgroup::Row& row, bool& isNull) override
+  {
+    return compiledOperator.compiled_function_int64(row.getData(), isNull);
+  }
+  double getDoubleVal(rowgroup::Row& row, bool& isNull) override
+  {
+    return compiledOperator.compiled_function_double(row.getData(), isNull);
+  }
+  float getFloatVal(rowgroup::Row& row, bool& isNull) override
+  {
+    return compiledOperator.compiled_function_float(row.getData(), isNull);
+  }
```

Comment:
Could we simply have one `compiled_function` member that will be `reinterpret_cast` to the appropriate type? That could avoid some run-time code duplication.